pkg/pillar/cmd/volumemgr/handlediskmetrics.go

```diff
@@ -270,6 +270,19 @@ func createOrUpdateDiskMetrics(ctx *volumemgrContext, wdName string) {
 		}
 	}
 
+	// XXX exclude ones already published above somehow
+	items, err := volumehandlers.GetAllDataSets(log, ctx)
```

Comment:
This is a generic code. I would expect here something like
```go
if handler := volumehandlers.GetZFSHandler(); handler != nil {
    items, err := handler.GetAlDataSets()
    ...
}
My point is `GetAlDataSets()` should be part of `handler`, not the entire `volumehandlers` package, as I see it as generic as possible.
```
PS. The current implementation still works; this comment is not critical for merging.